docs/proposals/clusterid/README.md

```diff
@@ -0,0 +1,118 @@
+
+### Goals
+
+- Set the cluster identifier for the cluster that has been joined to Karmada.
+
+### Non-Goals
+
+- The API describing cluster properties upgrade.
+
+## Proposal
+
+### User Stories (Optional)
+
+#### Story 1
+
+In Karmada, a cluster can be joined many times with different name. For a cluster joined Karmada repeatedly, once we remove one of them, all the clusters will turn into unready. More detail info can refer to [issue](https://github.com/karmada-io/karmada/issues/880).
+
+So we need to verify the cluster uniques when cluster joining.
+
+### Notes/Constraints/Caveats (Optional)
+
+### Risks and Mitigations
+
+## Design Details
+
+We can divide the implementation into two phases:
+
+- Using cluster's `kube-system` namespace uuid as the cluster id, and collect it into the [Cluster](https://github.com/karmada-io/karmada/blob/3039ebbba31358fc6d8a1a0374cef09dfc17cf5e/pkg/apis/cluster/v1alpha1/types.go#L26) object.
+- Adopting `ClusterProperty` CRD to identify a cluster in Karmada.
+
+There are two reasons for this: First, we want to fix the problem of repeat cluster join as soon as possible (preferably in v1.3); and second, we need to wait for the organization structure of Project [about-api](https://github.com/kubernetes-sigs/about-api) to be adjusted so that we can import the `ClusterProperty` API.
+
+### Phase one: collect kube-system ns uuid
+
+We propose to introduce an `ID` filed to the Cluster API:
+
+```go
+// ClusterSpec defines the desired state of a member cluster.
+type ClusterSpec struct {
+	// ID represents the cluster's unique identifier.
+    // +optional
+    ID string `json:"ID"`
+  
+	...
+}
+```
+
+During the cluster joining process, Karmada will just collect kubernetes cluster's `kube-system` namespace uuid as the cluster id.
+
+To ensure the uniqueness of the cluster in Karmada, we need to traverse all clusters that have been joined to Karmada and checkout whether their id are same as the cluster to be joined to Karmada. If they are the same, we will reject the join request, otherwise accept it.
+
+This check can be performed in two places: one in the join phase and the other in the Cluster object creation phase. We can choose one of them.
+
+### Phase two: adopt ClusterProperty API
+
+After waiting for the organization of the `about-io` project to be restructured, we can successfully import the `ClusterProperty` API.
+
+We propose to introduce a `ClusterProperty` filed to the Cluster API:
+
+```go
+// ClusterStatus contains information about the current status of a
+// cluster updated periodically by cluster controller.
+type ClusterStatus struct {
+    // ClusterProperty represents the cluster identity information.
+    // +optional
+    ClusterProperty map[string]string `json:"clusterProperty,omitempty"`
```

Comment:
Why is `map` ? I assume a cluster only has one property, right?

Reply:
there should be a string

Reply:
There may be multiple ClusterProperty with different names(such as `id.k8s.io`, `xx.karmada.io`, etc), we may need to collect them all.

Reply:
I'm not sure if we need to show cluster property in `Cluster` object, do you need to sync the changes made from the member cluster? 
I can't see much evidence for that, at least for now.